model_compression_toolkit/core/common/graph/virtual_activation_weights_node.py

```diff
@@ -0,0 +1,198 @@
+            input_shape: Input tensor shape of the node.
+            output_shape: Input tensor shape of the node.
+            weights: Dictionary from a variable name to the weights with that name in the layer the node represents.
+            layer_class: Class path of the layer this node represents.
+            reuse: Whether this node was duplicated and represents a reused layer.
+            reuse_group: Name of group of nodes from the same reused layer.
+            quantization_attr: Attributes the node holds regarding how it should be quantized.
+            has_activation: Whether the node has activations that we might want to quantize.
+            **kwargs: Additional arguments that can be passed but are not used (allows to init the object with an
+                existing node's __dict__).
+
+        """
+
+        super().__init__(name,
+                         framework_attr,
+                         input_shape,
+                         output_shape,
+                         weights,
+                         layer_class,
+                         reuse,
+                         reuse_group,
+                         quantization_attr,
+                         has_activation)
+
+        self.name = f"{VIRTUAL_ACTIVATION_WEIGHTS_NODE_PREFIX}_{act_node.name}_{weights_node.name}"
+
+        self.original_activation_node = act_node
+        self.original_weights_node = weights_node
+
+        v_candidates = []
+        for c_a in act_node.candidates_quantization_cfg:
+            for c_w in weights_node.candidates_quantization_cfg:
+                composed_candidate = CandidateNodeQuantizationConfig(activation_quantization_cfg=c_a.activation_quantization_cfg,
+                                                                     weights_quantization_cfg=c_w.weights_quantization_cfg)
+                v_candidates.append(composed_candidate)
+
+        # sorting the candidates by weights number of bits first and then by activation number of bits (reversed order)
+        v_candidates.sort(key=lambda c: (c.weights_quantization_cfg.weights_n_bits,
+                                         c.activation_quantization_cfg.activation_n_bits), reverse=True)
+
+        self.candidates_quantization_cfg = v_candidates
+
+    def get_bops_count(self, fw_impl: Any, fw_info: FrameworkInfo, candidate_idx: int) -> float:
+        """
+        Computes the composed node's (edge) bit-operation count.
+
+        Args:
+            fw_impl: A FrameworkImplementation object with framework specific methods.
+            fw_info: A FrameworkInfo object with framework specific information,
+            candidate_idx: The index of the node's quantization candidate configuration.
+
+        Returns: The BOPS count of the composed node.
+
+        """
+        node_mac = fw_impl.get_node_mac_operations(self.original_weights_node, fw_info)
+        node_bops = self.candidates_quantization_cfg[candidate_idx].weights_quantization_cfg.weights_n_bits * \
```

Comment:
The get_node_mac_operations is not in this PR (need to check other PRs).

How can we know the number of macs without knowing the node's type?

Reply:
you're right, get_bops_count needs to be added in the PR that follows this one